src/engine/GroupByHashMapOptimization.h

```diff
@@ -0,0 +1,137 @@
+// Copyright 2024, University of Freiburg,
+// Chair of Algorithms and Data Structures.
+// Author:
+//   2024      Fabian Krause (fabian.krause@students.uni-freiburg.de)
+
+#pragma once
+
+#include "engine/sparqlExpressions/RelationalExpressionHelpers.h"
+#include "engine/sparqlExpressions/SparqlExpressionValueGetters.h"
+
+struct AvgAggregationData {
+  using ValueGetter = sparqlExpression::detail::NumericValueGetter;
+  bool error_ = false;
+  double sum_ = 0;
+  int64_t count_ = 0;
+  void addValue(auto&& value, const sparqlExpression::EvaluationContext* ctx) {
+    auto val = ValueGetter{}(AD_FWD(value), ctx);
+    auto intValueAdder = [this](int64_t intValue) {
+      sum_ += static_cast<double>(intValue);
+    };
```

Comment:
This exact implementation also works for doubles, so if you
change the parameter to `auto`, or better
`[]<typename T>(T value) requires std::is_arithmetic_v<T> {... static_cast<double>};`
(unfortunately there is no `numeric` or `arithmetic` concept in the standard library.
Then you only need two overloads (numeric and non-numeric).